### DISCUSSION TITLE
How to build PCem?

### DISCUSSION THREAD

**Discussion by IBMPCXT5150** [PCemOnMac/PCemV17macOS#32] [Q&A] [answered]
I Am wondering how to build the PCem Source Code and I Would give many thanks if someone gives it in complied form or uncomplied form with instructions.

**Comment by almeath** [comment, marked as answer]
I built version 17 of PCem in late 2021, available here:

https://www.dropbox.com/s/y372f4jqj327y65/PCem17.zip?dl=0

Bear in mind, this was compiled on an Intel Mac running Mojave (10.14.6). I cannot vouch for how well it will work on newer versions of macOS or Apple Silicon machines. I am no longer developing on the Mac, so I cannot shed light on how to update the code to be more compatible with ARM Macs.

**Reply by IBMPCXT5150** [reply]
Many Thanks!! By the way, do you know where ROMS should be located? like ~/Library/Application Support/PCEM/roms

**Reply by almeath** [reply]
I cannot remember if this pre-compiled build looks for ROMs under ~/Library/Application Support/PCEM/roms , or if it was built before that change to the code, which means it may look for them inside the app bundle itself, under Contents/Resources/roms or something similar. I realize the latter does not follow standard macOS protocols, and on some systems it may trigger warnings about the app being "damaged". That kind of behavior is one of the reasons I moved away from macOS. Anyway, try inside the app bundle and see how you go, but I am sure the ROMs need to be inside a folder called "roms".

**Reply by almeath** [reply]
I remember there was discussion on this, and my preference was to keep my apps self-contained and keep all my support files inside the PCem app bundle, and others wanted to follow the macOS standard of putting everything external to the app in Application Support, so it is going to be one of those two alternatives ..

**Reply by IBMPCXT5150** [reply]
Actually, by your solution somewhere while searching for one, I Didn't know there was an another library in the usr directory. Many thanks again!!
